public/javascript/components/viewer.js

```diff
@@ -15,30 +16,64 @@ function updateViewer(viewportName, viewportConfig) {
         }
     }
 
+    if (viewportConfig.isReader) {
+        enableReader();
+        preventRefresh = true;
+    }
+
     currentViewPortConfig = viewportConfig;
     currentViewPortName = viewportName;
 
-    restyleViewer(isAnimated);
+    restyleViewer(isAnimated, preventRefresh);
 }
 
 function reloadiFrame() {
     viewerEl.src = viewerEl.src;
+
+    if (currentViewPortName === 'reader') {
+        enableReader();
+    }
 }
 
 function printViewer() {
     try {
-      viewerEl.contentWindow.print();
+        viewerEl.contentWindow.print();
+        analyticsCtrl.recordPrint();
     } catch (e) {
-      console.log("Can't communicate with iframe")
+        console.log('Can\'t communicate with iframe ', e);
     }
 }
 
-function restyleViewer(isAnimated) {
+function enableReader() {
+    try {
+        var printStyleSheets = viewerEl.contentDocument.querySelectorAll('link[media=\'print\']');
+
+        for (var i = 0; i < printStyleSheets.length; i++) {
+            printStyleSheets[i].setAttribute('media', 'all');
+        }
+
+        var styleLink = document.createElement('link');
+        styleLink.href = '/assets/styles/readerMode.css';
+        styleLink.rel = 'stylesheet';
+        styleLink.setAttribute('media', 'screen');
+        styleLink.type = 'text/css';
+
+        viewerEl.contentDocument.body.appendChild(styleLink);
+
+    } catch (e) {
+        console.log('Can\'t enable Reader mode: ', e);
```

Comment:
We should also try to provide some feedback to the user if they can't enter reader mode, such as an error bar, or making the button go red. Should be a minor edge case, such as when the user navigates away from a guardian page, so can be done later on.